backend/controller/dal/dal.go

```diff
@@ -84,8 +84,9 @@ func DeploymentArtefactFromProto(in *ftlv1.DeploymentArtefact) (DeploymentArtefa
 
 func runnerFromDB(row sql.GetRunnerRow) Runner {
 	var deployment optional.Option[model.DeploymentName]
-	if name, ok := row.DeploymentName.Get(); ok {
-		deployment = optional.Some(model.DeploymentName(name))
+	var zero model.DeploymentName
+	if row.DeploymentName != zero {
```

Comment:
From what I can tell, sqlc won't give us an optional on `RunnerRow.DeploymentName` because of the coalesce part of the sql statement?

This is also why parsing DeploymentName from "" is returning a DeploymentName without an error, but just a empty deployment name.

Is this all fine in Go or should we be more strict around using optionals vs zero?

Reply:
Wasn't this working before? Why would it change now?

Reply:
Fixed, optionals are maintained like before. And now parsing a deployment name of "" is an error